pkg/operator/controllers/azurensg/subnetnsg_test.go

```diff
@@ -0,0 +1,71 @@
+package azurensg
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-07-01/network"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/mock/gomock"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	utillog "github.com/Azure/ARO-RP/pkg/util/log"
+	mock_network "github.com/Azure/ARO-RP/pkg/util/mocks/azureclient/mgmt/network"
+)
+
+var (
+	vnet = azure.Resource{
+		SubscriptionID: "0000000-0000-0000-0000-000000000000",
+		ResourceGroup:  "vnet-rg",
+		ResourceName:   "vnet-name",
+	}
+)
+
+func TestFixSubnet(t *testing.T) {
+	r := AzureNSGReconciler{log: utillog.GetLogger()}
+	for _, tt := range []struct {
+		nsgname             string
+		architectureVersion api.ArchitectureVersion
+		expectedValue       string
+	}{
+		{
+			nsgname:             "someotherv1-nsg",
+			architectureVersion: api.ArchitectureVersionV1,
+			expectedValue:       "/subscriptions/0000000-0000-0000-0000-000000000000/resourceGroups/resourcegroup/providers/Microsoft.Network/networkSecurityGroups/infraid-node-nsg",
+		},
+		{
+			nsgname:             "infraid-nsg",
+			architectureVersion: api.ArchitectureVersionV2,
+			expectedValue:       "/subscriptions/0000000-0000-0000-0000-000000000000/resourceGroups/resourcegroup/providers/Microsoft.Network/networkSecurityGroups/infraid-nsg",
+		},
+		{
+			nsgname:             "someotherv2-nsg",
+			architectureVersion: api.ArchitectureVersionV2,
+			expectedValue:       "/subscriptions/0000000-0000-0000-0000-000000000000/resourceGroups/resourcegroup/providers/Microsoft.Network/networkSecurityGroups/infraid-nsg",
+		},
+	} {
```

Comment:
There is a convention already started within our tests to use t.run see e.g. https://github.com/Azure/ARO-RP/blob/master/pkg/operator/controllers/pullsecret/pullsecret_controller_test.go#L195 
With this principal when the test fails it signals which item in for loop failed, instead of a general test failure.

Reply:
+1 on using `t.Run`

Reply:
Put the test body in t.Run to enable parallel execution.